### ISSUE TITLE
Settings up NLog with VB.

### ISSUE THREAD

**Issue by ajtruckle** [NLog/NLog#3065]
I am using NLog for the first time with Visual Basic.

1. In your tutorial you suggest defining the logger like this:

`private static NLog.Logger logger = NLog.LogManager.GetCurrentClassLogger();`

But in Visual Basic I can't use the `Static` keyword:

`Private m_logger As NLog.Logger = LogManager.GetCurrentClassLogger()`

Please confirm.

2. Are there any DLL files that need to be deployed with the application that uses this logger?

3. Here is an example log entry:

2019-01-03 16:07:50.8120|ERROR|GoogleAuthandSync.Program|RESULT_FAILED_OAUTH

Why does it now actually show the exception content in any detail?

**Comment by snakefoot** [comment]
Using `static` is just an optimization, and not required. But in VB there should be a `shared`-keyword instead that should work just like `static`.

**Comment by ajtruckle** [comment]
@snakefoot I can't use `shared` either. I think the reason is because I have declared my logger variable in the "module" scope. I did this so that other methods would have access to the object.

If I declare it inside a method then I can use static. But that implies I have to declare the logging in every method. Confused.

**Comment by snakefoot** [comment]
Module scope should be the same as static. Again just an optimization not a requirement

Sent from my Sony Xperia


---- ajtruckle wrote ----


@snakefoot<https://github.com/snakefoot> I can't use shared either. I think the reason is because I have declared my logger variable in the "module" scope. I did this so that other methods would have access to the object.

If I declare it inside a method then I can use static. But that implies I have to declare the logging in every method. Confused.
